### PR TITLE
IGNITE-17466 Remove TableStorage and PartitionStorage implementations

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-17466

### CODE REVIEW

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java [hunk 1]
```diff
@@ -393,13 +392,30 @@ public void close() {
...
     }
 
+    /**
+     * Removes all data from this storage and frees all associated resources.
+     *
+     * @throws StorageException If failed to destroy the data or storage is already stopped.
+     */
+    public void destroy() {
+        try {
+            versionChainTree.destroy();
```
Comment (ibessonov) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java:402]: Why did you add this? This implementation makes no sense for persistent storage and is just wrong for volatile storage. And, it's synchronous. I don't like it.

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionlessLinks.java [hunk 2]
```diff
@@ -93,8 +93,4 @@ public static void writeToBuffer(ByteBuffer buffer, long link) {
 
         buffer.putInt(pageIndex(link));
     }
-
-    private PartitionlessLinks() {
```
Comment (ibessonov) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionlessLinks.java:97]: Oh my!

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/RowVersion.java [hunk 3]
```diff
@@ -142,38 +137,45 @@ boolean isCommitted() {
...
 
+    /** {@inheritDoc} */
     @Override
     public int headerSize() {
         return TIMESTAMP_STORE_SIZE_BYTES + NEXT_LINK_STORE_SIZE_BYTES + VALUE_SIZE_STORE_SIZE_BYTES;
     }
 
+    /** {@inheritDoc} */
     @Override
     public IoVersions<? extends AbstractDataPageIo> ioVersions() {
```
Comment (ibessonov) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/RowVersion.java:174]: Alright, since you modifying this class, may I ask you to change the signature to `public IoVersions<? extends AbstractDataPageIo<RowVersion>> ioVersions()...`? Or even remove a wildcard if it's possible

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionlessLinks.java (hunk 2 above)
Comment (tkalkirill) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PartitionlessLinks.java:97]: Yes

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/RowVersion.java (hunk 3 above)
Comment (tkalkirill) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/RowVersion.java:174]: Fix it

**Review by ibessonov — APPROVED** [review]
